### ISSUE TITLE
IDE0028 (Collection initialization can be simplified) results in invalid results with Span targets

### ISSUE THREAD

**Issue by christiannagel** [dotnet/roslyn#70101]
**Version Used**: 
Visual Studio 2022 Version 17.8.0 Preview 2
.NET 8.0.100-rc1.23455.8

**Steps to Reproduce**:

Using this code:

```csharp
char[] buffer = new char[4];
Span<char> span1 = buffer.AsSpan();
span1[0] = 'a';
Console.WriteLine(buffer[0]);
```

The result writes *a* to the console as expected, but this results in this warning:
IDE0028 Collection initialization can be simplified

Using the fix with the spread operator:

```csharp
char[] buffer = new char[4];
Span<char> span1 = [.. buffer];
span1[0] = 'a';
Console.WriteLine(buffer[0]);
```

still contains `0` in `buffer[0]`.

**Diagnostic Id**:
IDE0028

**Expected Behavior**:

I expect either using the spread to create a span referencing the buffer, or not to offer the IDE0028 warning in this case.

**Actual Behavior**:

Using the code fix, the compilation results in different behavior.
